sources/niLang/src/System_Linux.cpp

```diff
@@ -843,8 +845,8 @@ class cLinuxWindow : public ni::ImplRC<ni::iOSWindow,ni::eImplFlags_Default,ni::
   }
 
   virtual tBool __stdcall RedrawWindow() niImpl {
-    // TODO: IMPLEMENT
-    return eTrue;
+    dll_XClearArea(_GetDisplay(), _GetWindow(), 0, 0, 0, 0, True);
```

Comment:
I mean, it does the equivalent of the windows version :P If we are not sure if this is necessary (after all everything works without it) I will rollback and change the comment to "IMPLEMENT if necessary" :P

Reply:
My concern is the "XClear" naming, which is maybe horrible, but it implies that its going to do a "Clear", that is fill that rectangle with some background color. On Windows its "InvalidateRect" which 1) invalidates the rectangle (yes really) and 2) leads to a WM_PAINT message being generated. So the idea there is that you set the iOSWindow's refresh timer to 0 so that it doesnt "draw on update", it only draws on WM_PAINT and WM_PAINT is triggered manually with "RedrawWindow", this is useful to write UIs for tools where you dont want/need it to suck 100% of the CPU to just redraw the UI that hasnt changed.

To implement this correctly, as per the specs above you really have to test that whole interaction with the refresh timer, etc...

So anyway, my vote is just the TODO: IMPLEMENT comment since I dont think either of us knows exactly what this does. - which is what you've done so we'll keep it at that for now.